changelog.d/20230628_132155_chris_endpoint_setup_teardown.rst

```diff
@@ -0,0 +1,10 @@
+.. A new scriv changelog fragment.
+..
+.. Uncomment the header that is right (remove the leading dots).
+..
+New Functionality
+^^^^^^^^^^^^^^^^^
+
+- Added ``endpoint_setup`` and ``endpoint_teardown`` options to endpoint config, which,
+  if present, are run by the system shell during the endpoint initialization process and
+  shutdown process, respectively.
```

Comment:
Do you want to link to the documentation?  Could lead the documentation with what the section _will_ be (link-wise), or just link to the page.  Were it me, I might lean toward the former.